addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -5326,20 +5302,93 @@ registry.layout_column = SnippetOptionWidget.extend({
         }
         // Removing the grid properties.
         delete rowEl.dataset.rowCount;
-        rowEl.style.removeProperty('--grid-item-padding-x');
+        rowEl.style.removeProperty('--grid-item-padding-x'); // Kept for compatibility.
         rowEl.style.removeProperty('--grid-item-padding-y');
     },
+    /**
+     * Removes all the eventual grid previews, to avoid unwanted behaviors (when
+     * toggling normal mode for example).
+     *
+     * @private
+     */
+    _clearAllPreviews(columnEls) {
+        // Removing the padding previews.
+        for (const columnEl of columnEls) {
+            if (columnEl._removePaddingPreview) {
+                columnEl._removePaddingPreview();
+            }
+        }
+    },
+});
+
+registry.GridColumns = SnippetOptionWidget.extend({
+    /**
+     * @override
+     */
+    cleanForSave() {
+        // Remove the padding highlights.
+        if (this.$target[0].classList.contains("o_we_padding_highlight")) {
+            this._removePaddingPreview();
+        }
+    },
+    /**
+     * @override
+     */
+    cleanUI() {
+        // Remove the padding highlights.
+        if (this.$target[0].classList.contains("o_we_padding_highlight")) {
+            this._removePaddingPreview();
+        }
+    },
```

Comment:
Maybe we should call `cleanUI` everywhere `cleanForSave`  is already called? (maybe just calling cleanUI` in the `SnippetEditor` `cleanForSave` would be enough?)

Reply:
Yes, I didn't do it because you said to just do it for the clone first, so I thought it could stay like this for the time being and that it would be done later in another PR (where the duplicate would be removed) :open_mouth: 

I will do that then :slightly_smiling_face:

Reply:
Yes, what I meant was that you did not need to check all the cleanForSave definition to extract the relevant parts into cleanUI for now. But definitely call cleanUI at the same time as cleanForSave :+1: